### PR TITLE
Clarify iOS app use cases

### PR DESCRIPTION
**Pull request by dmipon**
Add example which clarifies VoIP certificate upload

### CODE REVIEW

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 1]
```diff
@@ -98,6 +98,29 @@ Your push notification certificates are now uploaded and ready for use.
...
+1. Multiple Sinch Applications, one iOS app (one Bundle ID)
+
+#### Single Sinch Application, multiple iOS apps (i.e. multiple different Bundle IDs)
+
+Suppose you have a patient app and a doctor app with `PatBundleID` and `DocBundleID` respectively.
+You want a doctor to be able to call a patient. You need to generate *VoIP certificate* for the app with `PatBundleID` and upload it to *Sinch Developer Portal* for the *Sinch Application* which key and secret you are going to use in both iOS apps.
+
+> ⚠️ Note that both iOS apps should be signed using the same signing certificate either *Apple Development Certificate* or *iOS Distribution Certificate*. Chosen certificate must match ANPS environment setting provided to SinchSDK. If you sign using *Apple Development Certificate* please provide `SINAPSEnvironmentDevelopment` to Sinch SDK. If you sign using *iOS Distribution Certificate* please provide `SINAPSEnvironmentProduction`.
+
+Miss match in APNS environment settings and signing identity in one or both iOS apps will result in ability to make a call from the doctor app to the patient app.
```
Comment (GiovanniBeriSinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:115]: typo: 
```suggestion
Mismatch in APNS environment settings and signing identity in one or both iOS apps will result in ability to make a call from the doctor app to the patient app.
```

Comment (GiovanniBeriSinch) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:115]: I guess, typo:
```suggestion
Miss match in APNS environment settings and signing identity in one or both iOS apps will result in inability to make a call from the doctor app to the patient app.
```

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 2]
```diff
@@ -98,6 +98,29 @@ Your push notification certificates are now uploaded and ready for use.
...
+
+> ⚠️ Note that both iOS apps should be signed using either *Apple Development Certificate* or *iOS Distribution Certificate*. Chosen certificate must match ANPS environment setting provided to Sinch SDK. If you sign using *Apple Development Certificate* please provide `SINAPSEnvironmentDevelopment` to Sinch SDK when SINManagedPush is created. If you sign using *iOS Distribution Certificate* please provide `SINAPSEnvironmentProduction` when SINManagedPush is created.
+
+Miss match in APNS environment settings and signing identity in one or both iOS apps will result in ability to make a call from the doctor app to the patient app.
+
+#### Multiple Sinch Applications, one iOS app (one Bundle ID)
+
+Suppose that we have two Sinch Apps, SApp1 and SApp2 respectively and only one iOS app. If you want to make a call from iOS app with SApp1 to iOS app with SApp2 you need to generate VoIP certificate for iOS app bundle ID and upload it to the caller app in *Sinch Developer Portal*. In our example you will upload VoIP certificate to SApp1 in the portal.
```
Comment (GiovanniBeriSinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:119]: It would be nice if you had a real life example of this scenario, in a similar fashion to the doctor/patient app of the previous case

Comment (GiovanniBeriSinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:119]: > generate VoIP certificate for iOS app bundle ID and upload it to the caller app in *Sinch Developer Portal*

in the doc/patient scenario you mention before, you wrote that the cert has to be uploaded for the patient app i.e. the **callee** app, while here you mention it should be done for the **caller** app.

Is it intended? Am I missing something?

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 3]
```diff
@@ -98,6 +98,29 @@ Your push notification certificates are now uploaded and ready for use.
...
+
+1. Single Sinch Application, multiple iOS apps (i.e. multiple different Bundle IDs)
+1. Multiple Sinch Applications, one iOS app (one Bundle ID)
+
+#### Single Sinch Application, multiple iOS apps (i.e. multiple different Bundle IDs)
+
+Suppose you have a patient app and a doctor app with `PatBundleID` and `DocBundleID` respectively.
+You want a doctor to be able to call a patient. You need to generate *VoIP certificate* for the app with `PatBundleID` and upload it to *Sinch Developer Portal* for the *Sinch Application* which key and secret you are going to use in both iOS apps.
+
+> ⚠️ Note that both iOS apps should be signed using either *Apple Development Certificate* or *iOS Distribution Certificate*. Chosen certificate must match ANPS environment setting provided to Sinch SDK. If you sign using *Apple Development Certificate* please provide `SINAPSEnvironmentDevelopment` to Sinch SDK when SINManagedPush is created. If you sign using *iOS Distribution Certificate* please provide `SINAPSEnvironmentProduction` when SINManagedPush is created.
```
Comment (GiovanniBeriSinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:113]: I'd reword it like this:
```suggestion
> ⚠️ Note that both iOS apps should be signed using either *Apple Development Certificate* or *iOS Distribution Certificate*. Chosen certificate must match ANPS environment setting provided to Sinch SDK when `SINManagedPush` is created: if you sign using *Apple Development Certificate* please provide `SINAPSEnvironmentDevelopment`, if you sign using *iOS Distribution Certificate* please provide `SINAPSEnvironmentProduction`.
```

**Review by GiovanniBeriSinch — CHANGES_REQUESTED** [review]
some minor rewording/typos, and a more relevant question about voip certs

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md (hunk 2 above)
Comment (dmipon) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:119]: In the first case Sinch APP is the only one and it matters for which bundle ID we generate the VoIP certificate. In the second case bundle ID is the one and it matters into which Sinch APP we upload the certificate

Comment (christoffer-a-sinch) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md]: I'm myself not so keen on using the doctor/patient concrete example, I think I'd prefer just _A_ and _B_... But maybe more importantly we should at least be consistent in the two examples. 

When referring to "Sinch Apps", use _Sinch Applications_ . E.g. we tend to spell out full _Application_ in when referencing a _Sinch Application_, but use just "app" if speaking of very generic apps like "iOS app".

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 4]
```diff
@@ -98,6 +98,30 @@ Your push notification certificates are now uploaded and ready for use.
...
+Suppose you have a patient app and a doctor app with `PatBundleID` and `DocBundleID` respectively.
+You want a doctor to be able to call a patient. You need to generate *VoIP certificate* for the app with `PatBundleID` and upload it to *Sinch Developer Portal* for the *Sinch Application* which key and secret you are going to use in both iOS apps.
+
+> ⚠️ Note that both iOS apps should be signed using either *Apple Development Certificate* or *iOS Distribution Certificate*. Chosen certificate must match ANPS environment setting provided to Sinch SDK when `SINManagedPush` is created: if you sign using *Apple Development Certificate* please provide `SINAPSEnvironmentDevelopment`, if you sign using *iOS Distribution Certificate* please provide `SINAPSEnvironmentProduction`.
+
+Mismatch in APNS environment settings and signing identity in one or both iOS apps will result in inability to make a call from the doctor app to the patient app.
+
+#### Multiple Sinch Applications, one iOS app (one Bundle ID)
+
+Suppose that we have two Sinch Apps (in Sinch Portal), Driver App and Passenger App and only one iOS app.
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:119]: To a customer that is not in the ride-hailing segment I think this "driver" and "rider" examples will actually not make sense. I think we should make it generic and only speak of "Application A" and "Application B". And be consistent with that in both examples.

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 5]
```diff
@@ -98,6 +98,30 @@ Your push notification certificates are now uploaded and ready for use.
...
+
+Depending on your use case it is possible to configure calling functionality differently. Let's consider two possible scenarios:
+
+1. Single Sinch Application, multiple iOS apps (i.e. multiple different Bundle IDs)
+1. Multiple Sinch Applications, one iOS app (one Bundle ID)
+
+#### Single Sinch Application, multiple iOS apps (i.e. multiple different Bundle IDs)
+
+Suppose you have a patient app and a doctor app with `PatBundleID` and `DocBundleID` respectively.
+You want a doctor to be able to call a patient. You need to generate *VoIP certificate* for the app with `PatBundleID` and upload it to *Sinch Developer Portal* for the *Sinch Application* which key and secret you are going to use in both iOS apps.
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:111]: It is only in the 4.x SDK we had a factory method that allowed injecting secret. In the 5.x sample apps we only allow JWT auth. Though the secret is technically in the sample app, is not intended like that in production deployments. So, I think we should just remove the reference to `secret` here and only speak of _Sinch Application_ and _Application Key_.

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 6]
```diff
@@ -98,6 +98,29 @@ Your push notification certificates are now uploaded and ready for use.
 
 Certificates configured with Sinch can be replaced or renewed by uploading new ones. New certificates will automatically replace the previous ones for their respective type (_Development_, _Production_ and _VoIP Services_).
 
+### Possible scenarios of configuring your application
+
+Depending on your use case it is possible to configure calling functionality differently. Let's consider two possible scenarios:
+
+1. Single Sinch Application, multiple iOS apps (i.e. multiple different Bundle IDs)
+1. Multiple Sinch Applications, one iOS app (one Bundle ID)
```
[the remark this answers is not in the document]
Comment (dmipon) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:106]: Nope. Markdown renderer will display it correctly regardless of what is here. It's a feature to avoid confusion when add/remove/change positions of items

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md (hunk 4 above)
Comment (dmipon) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:119]: Heh :) okay i'll fix it back again :)

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md [hunk 7]
```diff
@@ -98,6 +98,30 @@ Your push notification certificates are now uploaded and ready for use.
...
+You want App-B to be able to call App-A. You need to generate *VoIP certificate* for the app with `A-BundleID` and upload it to *Sinch Developer Portal* for the *Sinch Application* which application key you are going to use in both iOS apps.
+
+> ⚠️ Note that both iOS apps should be signed using either *Apple Development Certificate* or *iOS Distribution Certificate*. Chosen certificate must match ANPS environment setting provided to Sinch SDK when `SINManagedPush` is created: if you sign using *Apple Development Certificate* please provide `SINAPSEnvironmentDevelopment`, if you sign using *iOS Distribution Certificate* please provide `SINAPSEnvironmentProduction`.
+
+Mismatch in APNS environment settings and signing identity in one or both iOS apps will result in inability to make a call from App-B to App-A.
+
+#### Multiple Sinch Applications, one iOS app (one Bundle ID)
+
+Suppose that we have two Sinch Apps (in Sinch Portal), App-A and App-B and only one iOS app.
+iOS app could be build with App-A application key or App-B application key.
```
Comment (GiovanniBeriSinch) [inline docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md:120]: typo
```suggestion
iOS app could be built with App-A application key or App-B application key.
```

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md (hunk 2 above)
Comment (GiovanniBeriSinch) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md]: so, is the general rule:
"the **caller Sinch App** needs to have a certificate of the **callee iOS app**"
?

I trust the comments you wrote are correct, I am just trying to understand it myself 😃

**Review by GiovanniBeriSinch — CHANGES_REQUESTED** [review]
typo

File: docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md (hunk 2 above)
Comment (dmipon) [reply docs/voice/voice-for-ios/voice-ios-push-notifications-callkit.md]: yes
